File: Easy/29Semordnilap.ts
function semordnilap(words) {
  // Write your code here.
  let wordSet = new Set(words);
  let semordnilapPairs: string[][] = [];
  //   let semordnilapPairs = [];

  for (let word of words) {
    let reversed = word.split("").reverse().join("");
    if (wordSet.has(reversed) && reversed != word) {
      semordnilapPairs.push([word, reversed]);
      wordSet.delete(word);
      wordSet.delete(reversed);
    }
  }
  return semordnilapPairs;
}

console.log("====================================");
console.log(semordnilap(["dog", "dog", "desserts", "god", "stressed"]));
console.log("====================================");

File: Easy/29Semordnilap.py
def semordnilap(words):
    # Write your code here.
    wordSet = set(words)
    semordnilapPairs = []
    for word in words:
        reverse = word[::-1]
        if reverse in wordSet and word != reverse:
            semordnilapPairs.append([word, reverse])
            wordSet.remove(reverse)
            wordSet.remove(word)

    return semordnilapPairs


print(semordnilap(["dog", "dog", "desserts", "god", "stressed"]))
